### ISSUE TITLE
How to change code if target is a seq other than a word?

### ISSUE THREAD

**Issue by ylqfp** [mranahmd/dssm-wemb-theano#1]
Hi, I'm trying to reimplement your code DSSM in query-doc screenario. However, I found that your code is not suitable in my screenario, for my dataset is in this form <query, doctitle1:click, doctitle2:click, doctitle3:noclick...>
So I need to construct a mapping from query -> doc, However, I don't know how to change your code, would you plz help me? 
Many Thanks!!!

**Comment by mranahmd** [comment]
Hi,

Yes the code is written for learning word embeddings with a context-word dataset and it cannot be applied directly to query-doc setup. However it should not difficult to change the code for query-doc setup. We can do it.

Do you already have both positive and negative document samples for each query?

Best Regards,

**Comment by ylqfp** [comment]
Dear Mranahmd,
So glad to get your reply!
Yes! My dataset is like this:
[
query1 \t doc21:1 \t doc2:1 \t doc13:1 \t ... doc32:0 ...doc3:0 ,
query2 \t ...
...
]
Here, 1 stands for "Related", 0 stands for "Not Related".
Currently, I'm using Keras to transform into a binary classification problem, however, the loss is not Ranking loss, which may bring performence loss.

**Comment by ylqfp** [comment]
Query -> LSTM -> embedding1
Doc -> LSTM -> embedding2
cos(embedding1, embedding2) -> sigmoid
This is how I'm done currently.

**Comment by mranahmd** [comment]
If you are using keras then the existing DSSM code may not be very useful. 
And I would suggest looking at the mnist_siamese_graph.py example of keras.
The DSSM is very similar to a siamese network.

If you will be using theano then after you computed the LSTM embeddings you can do the following to implement the similarity part:
[] LSTM embedding of the queries in a mini-batch go into projQ
[] LSTM embedding of the positive documents go into embD
[] LSTM embedding of the negative documents go into embDNeg

Best,

**Comment by ylqfp** [comment]
Thanks! I really appreciate your help!
I'll try keras's siamese graph first, and then try to rewrite dssm in theano.
